test_integration/02_stake_and_mint/utils/progress_stake_assertion.js

```diff
@@ -208,14 +212,22 @@ class ProgressStakeAssertion {
     );
     assert.strictEqual(
       eventData._proofProgress,
-      false,
-      'Proof progress flag should be false.',
-    );
-    assert.strictEqual(
-      eventData._unlockSecret,
-      stakeRequest.unlockSecret,
-      'Unlock secret must match.',
+      proofProgress,
+      `Proof progress flag should be ${proofProgress}.`,
     );
+    if (proofProgress === true) {
+      assert.strictEqual(
+        eventData._unlockSecret,
+        Utils.ZERO_BYTES32,
```

Comment:
Same as above. Maybe a comment could explain it?

Reply:
Added